src/Adapter/MySQL.php

```diff
@@ -336,6 +337,37 @@ private function computeOrderByField(array $filterToTableMapping)
             $orderField = 'p.' . $orderField;
         }
 
+        // put some products at the end of the list
+        $orderField = $this->computeShowLast($orderField);
+
+        return $orderField;
+    }
+
+    /**
+     * Sort product list: InStock, OOPS with qty 0, OutOfStock
+     * @param  string $orderField
+     * @return string
+     */
+    private function computeShowLast($orderField)
+    {
+        // allow only if feature is enabled
+        if(! \Configuration::get('PS_LAYERED_FILTER_SHOW_OUT_OF_STOCK_LAST')) {
+            return $orderField;
+        }
+
+        if ($this->getInitialPopulation() !== null && !empty($orderField)) {
```

Comment:
You can compute both if
```suggestion
        if ($this->getInitialPopulation() === null 
            || empty($orderField) 
            || !Configuration::get('PS_LAYERED_FILTER_SHOW_OUT_OF_STOCK_LAST')
        ) {
```

Reply:
Changed as you suggested. Commit:  https://github.com/PrestaShop/ps_facetedsearch/pull/154/commits/5f66894a8787a6ba978cd7a2510893aeceea2761